File: SW Expert Academy/D2/02001. 파리 퇴치/파리 퇴치.py
for tc in range(int(input())):
    N,M = map(int, input().split())
    flies = []
    for i in range(N):
        flies.append(list(map(int, input().split())))

    print("#%d"%(tc+1), end=' ')

    max_fly = 0
    for i in range(N-M+1):
        for j in range(N-M+1):
            sum_fly = 0
            for x in range(i, i+M):
                sum_fly += sum(flies[x][j:j+M])

            max_fly = max(sum_fly, max_fly)

    print(max_fly)

File: SW Expert Academy/D2/02001. 파리 퇴치/파리 퇴치.cpp
#include <iostream>
#include <vector>
using namespace std;

int main(){
	int T; cin>>T;
	for(int tc=0; tc<T; tc++){
		int N,M; cin>>N>>M;
		vector<vector<int>> flies;
		for(int i=0; i<N; i++){
			vector<int> v(N);
			for(int j=0; j<N; cin>>v[j++]);
			flies.push_back(v);
		}
		
		printf("#%d ", tc+1);
		
		int max_fly = 0;
		for(int i=0; i<=N-M; i++){
			for(int j=0; j<=N-M; j++){
				int sum_fly = 0;
				for(int x=i; x<i+M; x++)
					for(int y=j; y<j+M; sum_fly+=flies[x][y++]);
				
				if(sum_fly > max_fly) max_fly = sum_fly;
			}
		}
		
		printf("%d\n", max_fly);
	}
}
